src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -546,6 +546,105 @@ def another_user(user=None, tenant=None, account=None):
                       calling_format='boto.s3.connection.OrdinaryCallingFormat')
     return conn, arn
 
+
+def list_topics_by_admin(assert_len, tenant=''):
+    if tenant == '':
+        result = admin(['topic', 'list'], get_config_cluster())
+    else:
+        result = admin(['topic', 'list', '--tenant', tenant], get_config_cluster())
+    parsed_result = json.loads(result[0])
+    assert_equal(len(parsed_result['topics']), assert_len)
+    return parsed_result
+
+
+def get_stats_persistent_topic_by_admin(topic_name, expected_entries_number):
+    result = admin(['topic', 'stats', '--topic', topic_name], get_config_cluster())
+    assert_equal(result[1], 0)
+    parsed_result = json.loads(result[0])
+    assert_equal(parsed_result['Topic Stats']['Entries'], expected_entries_number)
+    return parsed_result
+
+
+def get_topic_by_admin(topic_name, tenant=''):
+    if tenant == '':
+        result = admin(['topic', 'get', '--topic', topic_name], get_config_cluster())
+    else:
+        result = admin(['topic', 'get', '--topic', topic_name, '--tenant', tenant], get_config_cluster())
+    assert_equal(result[1], 0)
+    parsed_result = json.loads(result[0])
+    return parsed_result
+
+
+def remove_topic_by_admin(topic_name, tenant='', expected_failure=False):
+    if tenant == '':
+        result = admin(['topic', 'rm', '--topic', topic_name], get_config_cluster())
+    else:
+        result = admin(['topic', 'rm', '--topic', topic_name, '--tenant', tenant], get_config_cluster())
+    if not expected_failure:
+        assert_equal(result[1], 0)
```

Comment:
there are 2 situation for an expected failure:
1. removing removed or non-existent topic
2. realm is reloading and we are polling on the remove operation

Reply:
my question is, what if we expect failure, but the operation did not fail. shouldn't we assert in this case?

Reply:
in case we remove non-existent topic, this would be true
but in case of polling while realm reloading, it should succeed eventually
maybe we should leave it up to the test

Reply:
ok. so, maybe instead of "expect_failure" whic himplies that you should assert on "success".
you should call this paramter "allow_failure" - this means that you allow failures to happen, at least for some time?